sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_algorithms.asciidoc

```diff
@@ -0,0 +1,268 @@
+
+|+template <typename Group, typename T, class Predicate> bool all_of(Group g, T x, Predicate pred);+
+|Return +true+ if _pred(x)_ is +true+ for all work-items in the group. _pred_ must be the same for all work-items in the group.
+
+|+template <typename Group, typename T, class Predicate> bool none_of(Group g, T x, Predicate pred);+
+|Return +true+ if _pred(x)_ is +true+ for no work-items in the group. _pred_ must be the same for all work-items in the group.
+|===
+
+|===
+|Function|Description
+
+|+template <typename Group, typename Ptr, class Predicate> bool any_of(Group g, Ptr first, Ptr last, Predicate pred);+
+|Return +true+ if _pred_ returns +true+ for any element in the range [_first_, _last_). _first_, _last_ and _pred_ must be the same for all work-items in the group.
+
+|+template <typename Group, typename Ptr, class Predicate> bool all_of(Group g, Ptr first, Ptr last, Predicate pred);+
+|Return +true+ if _pred_ returns +true+ for all elements in the range [_first_, _last_). _first_, _last_ and _pred_ must be the same for all work-items in the group.
+
+|+template <typename Group, typename Ptr, class Predicate> bool none_of(Group g, Ptr first, Ptr last, Predicate pred);+
+|Return +true+ if _pred_ returns +true+ for no element in the range [_first_, _last_). _first_, _last_ and _pred_ must be the same for all work-items in the group.
+|===
+
+==== Collectives
+
+In this section, the meaning of "exclusive scan" and "inclusive scan" are as defined in Sections 29.8.7 and 29.8.8 of the {cpp}17 specification, respectively.
+
+|===
+|Function|Description
+
+|+template <typename Group, typename T> T broadcast(Group g, T x);+
+|Broadcast the value of _x_ from the work-item with the lowest id to all work-items within the group.
+
+|+template <typename Group, typename T> T broadcast(Group g, T x, Group::linear_id_type local_linear_id);+
+|Broadcast the value of _x_ from the work-item with the specified linear id to all work-items within the group.  The value of _local_linear_id_ must be the same for all work-items in the group.
+
+|+template <typename Group, typename T> T broadcast(Group g, T x, Group::id_type local_id);+
+|Broadcast the value of _x_ from the work-item with the specified id to all work-items within the group. The value of _local_id_ must be the same for all work-items in the group, and its dimensionality must match the dimensionality of the group.
+
+|+template <typename Group, typename T, class BinaryOperation> T reduce(Group g, T x, BinaryOperation binary_op);+
+|Combine the values of _x_ from all work-items in the group using the operator _binary_op_, which must be one of the group algorithms library function objects.  _binary_op_ must be the same for all work-items in the group.
+
+|+template <typename Group, typename V, typename T, class BinaryOperation> T reduce(Group g, V x, T init, BinaryOperation binary_op);+
```

Comment:
Do we want to deduce `V` and `T` separately and use `T` as return type? Also it doesn't say what the accumulator type is. I suggest we keep it simple and don't allow mixed type to be automatic. Meaning only deduce once (like e.g. `std::max`) and also require that `is_same_v<decltype(binary_op(x,x)), T>`. If we keep it simple like that then we should make it easy to specify the type and should make `T` the first template argument.

Reply:
I've attempted to address this in 4ee810.  Is the new text clear?